velox/exec/tests/GroupedExecutionTest.cpp

```diff
@@ -283,6 +284,150 @@ TEST_F(GroupedExecutionTest, groupedExecutionWithOutputBuffer) {
   EXPECT_EQ(18, taskStats.pipelineStats[1].operatorStats[1].inputVectors);
 }
 
+DEBUG_ONLY_TEST_F(
+    GroupedExecutionTest,
+    groupedExecutionWithHashJoinSpillCheck) {
+  // Create source file to read as split input.
+  auto vectors = makeVectors(24, 20);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->path, vectors);
+
+  struct {
+    bool enableSpill;
+    bool mixedExecutionMode;
+    int expectedNumDrivers;
+    bool expectedSpill;
+
+    std::string debugString() const {
+      return fmt::format(
+          "enableSpill {}, mixedExecutionMode {}, expectedNumDrivers {}, expectedSpill {}",
+          enableSpill,
+          mixedExecutionMode,
+          expectedNumDrivers,
+          expectedSpill);
+    }
+  } testSettings[] = {
+      {false, false, 12, false},
+      {false, true, 9, false},
+      {true, false, 12, true},
+      {true, true, 9, false}};
+
+  for (const auto& testData : testSettings) {
+    SCOPED_TRACE(testData.debugString());
+
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId probeScanNodeId;
+    core::PlanNodeId buildScanNodeId;
+
+    PlanBuilder planBuilder(planNodeIdGenerator, pool_.get());
+    planBuilder.tableScan(rowType_)
+        .capturePlanNodeId(probeScanNodeId)
+        .project({"c3 as x", "c2 as y", "c1 as z", "c0 as w", "c4", "c5"});
+    // Hash join.
+    core::PlanNodeId joinNodeId;
+    planBuilder
+        .hashJoin(
+            {"w"},
+            {"r"},
+            PlanBuilder(planNodeIdGenerator, pool_.get())
+                .tableScan(rowType_, {"c0 > 0"})
+                .capturePlanNodeId(buildScanNodeId)
+                .project({"c0 as r"})
+                .planNode(),
+            "",
+            {"x", "y", "z", "w", "c4", "c5"})
+        .capturePlanNodeId(joinNodeId)
+        .project({"w as c0", "z as c1", "y as c2", "x as c3", "c4", "c5"})
+        .planNode();
+    auto planFragment =
+        planBuilder
+            .partitionedOutput({}, 1, {"c0", "c1", "c2", "c3", "c4", "c5"})
```

Comment:
Why do we need this partitionedOutput node?

Reply:
This is used for output for a async task execution.